### ISSUE TITLE
Flaky tests on Appveyor

### ISSUE THREAD

**Issue by gabro** [scalameta/metals#435]
Most Appveyor builds fail due flaky sbt tests.
I'm not sure of the root cause, but I think we're slipping into ignoring those build results by default (at least I am) so it would be good to fix it and start trusting the Windows build again

**Comment by PanAeon** [comment]
Hi,
I tried to run the build on appveyor from my fork. It built version `0.3.3+90-ef650105-SNAPSHOT` probably because of `dynVersion`.
I tried to RDP into the build machine (which just worked, btw), fortunately at the same time my built went red. What I found is quite high memory usage:

<img width="1108" alt="screen shot 2019-01-04 at 15 31 18pm" src="https://user-images.githubusercontent.com/686076/50698836-02a69880-103e-11e9-9802-f748ee0896b9.png">

The built starts sbt, bloop, and it looks like it spawns a child sbt process. Right now it sets `-Xmx4G` in the `.jvmopts`. The machine I've been allocated has got 4Gb of memory, plus 1.5G of a page file, so obviously it can't accomodate 3*4Gb memory, allocated for the java processes. 
So I suspect that bloop's JVM asks OS for too much memory, and gets paged. Then for the sbt client it appears as bloop has hanged and it starts another process / timeouts.
I'm not really sure what to do with this, as setting memory too low will also result in "jvm freezing", also scala compiler sometimes require quite a bit of ram.
I think it might be a good idea to adjust `Xmx` to at most `2Gb`  and see how the build is going to behave.

**Comment by olafurpg** [comment]
Great job investigating @PanAeon !

We definitely don't need that much memory for CI, the `.jvmopts` has those large values because sbt consumes a lot of memory with frequent `reload` and long-running `~test`. Too large `-Xmx` settings don't cause so much problems on Linux/macOS compared to Windows in my experience.

Should we try to overwrite `.jvmopts` in appveyor to use smaller memory settings?

**Comment by olafurpg** [comment]
I'm gonna closed this as fixed thanks to the contributions by @PanAeon! We should still keep an eye for flaky Windows tests since it's not guaranteed all issues have been fixed.

**Comment by olafurpg** [comment]
Retries observed on Travis CI https://ci.appveyor.com/project/scalameta/metals/builds/21462999

```
INFO  sbt exit: 0
INFO  time: ran 'sbt bloopInstall' in 1m39s
INFO  running embedded 'bloop bsp --protocol tcp --host 127.0.0.1 --port 1165'
WARN  BSP retry 1
INFO  running embedded 'bloop bsp --protocol tcp --host 127.0.0.1 --port 1168'
WARN  BSP retry 2
INFO  running embedded 'bloop bsp --protocol tcp --host 127.0.0.1 --port 1171'
WARN  BSP retry 3
INFO  running embedded 'bloop bsp --protocol tcp --host 127.0.0.1 --port 1174'
WARN  BSP retry 4
INFO  running embedded 'bloop bsp --protocol tcp --host 127.0.0.1 --port 1177'
INFO  tracing is disabled for protocol BSP, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\appveyor\AppData\Local\scalameta\metals\cache\bsp.trace.json
ERROR Timeout waiting for 'build/initialize' response
WARN  BSP retry 5

```

**Comment by olafurpg** [comment]
Fixed in #460. Let's see if that did the trick 😏

**Comment by olafurpg** [comment]
So far haven't seen any flaky behavior on Appveyor after disabling `SbtSlowSuite`. It's also gotten a lot faster, completes in ~8 minutes now of which 2 minutes are before the first `sbt publishLocal`
